src/QuartoNotebookWorker/src/QuartoNotebookWorker.jl

```diff
@@ -0,0 +1,31 @@
+module QuartoNotebookWorker
+
+module Packages
+
+import TOML
+
+is_precompiling() = ccall(:jl_generating_output, Cint, ()) == 1
+
+const packages = let
+    project = TOML.parsefile(joinpath(@__DIR__, "..", "Project.toml"))
+    uuid = Base.UUID(project["uuid"])
+    key = "packages"
+    is_precompiling() && Base.record_compiletime_preference(uuid, key)
```

Comment:
This is the key part that allows this to, it seems so far, be a usable approach both for user and developer.

In `QNR` at precompilation time we find all the packages we want to vendor into the worker and serialize their source. During `QNR` runtime module init we move that vendored source to version-keyed scratch spaces that we record as preferences that are then passed to the `QNW` during it's precompilation. This then loads the right vendored source into `QNW`. If the vendored deps change, this all recompiles as expected. If we then load up `QNW` we can still use `Revise`, `Debugger`, etc. to easily debug and develop code within the worker code without having to deal with restarts.